blocks/amp-story/amp-story-grid-layer.js

```diff
@@ -0,0 +1,124 @@
+const { __ } = wp.i18n;
+const {
+	registerBlockType
+} = wp.blocks;
+const {
+	InspectorControls,
+	InnerBlocks
+} = wp.editor;
+const {
+	SelectControl
+} = wp.components;
+
+const ALLOWED_BLOCKS = [
+	'core/code',
+	'core/embed',
+	'core/image',
+	'core/list',
+	'core/paragraph',
+	'core/preformatted',
+	'core/pullquote',
+	'core/quote',
+	'core/table',
+	'core/verse',
+	'core/video'
+];
+
+function setBlockParent( props ) {
+	if ( ALLOWED_BLOCKS.includes( props.name ) ) {
+		return Object.assign(
+			{},
+			props,
+			{ parent: [ 'amp/amp-story-grid-layer' ] }
+		);
+	}
+	return props;
+}
+
+wp.hooks.addFilter(
+	'blocks.registerBlockType',
+	'amp/set-block-parents',
+	setBlockParent
+);
+
+// @todo Perhaps this should wait behind https://github.com/WordPress/gutenberg/issues/8589?
```

Comment:
Commented this out since with using `allowedBlocks` param for the inner blocks it should already restrict the usage of the blocks. Since only `amp-story-page` should be allowed on the root level, perhaps it wouldn't make sense to unregister other blocks here? Maybe we should wait for https://github.com/WordPress/gutenberg/issues/8589 to be able to use `amp-story-page` as the default instead.